src/helpers/domainFindOptionsHelper.ts

```diff
@@ -0,0 +1,136 @@
+import { HttpRequest } from "@azure/functions";
+import { SortDirection } from "mongodb";
+import { DomainFindOptions } from "@zero-tech/data-store-core";
+import { DomainSortDirection, domainReflectionSchema } from "../types";
+declare type AscDesc = -1 | 1;
+declare type IncludeExlucde = 0 | 1;
+interface dynamicObject {
+  [key: string]: any;
+}
+const defaultProjection = ["domainid", "owner", "name"];
+const defaultSort = ["label", "owner"];
+const defaultSortDirection: AscDesc[] = [-1, -1];
+
+/*
+ * Generates a DomainFindOptions object based on query string parameters
+ * @param req - An HttpRequest
+ * @returns  - a DomainFindOptions object, with either default values or values supplied from the query string
+ */
+export function getDomainFindOptionsFromQuery(req: HttpRequest) {
+  //Get variables to project, matching the case insensitive names to case sensitive object names
+  const projectionObject = createProjection(req);
+  const sortObject = createSort(req);
+
+  let findOptions: DomainFindOptions = {
+    sort: sortObject,
+    projection: projectionObject,
+    limit: isNaN(+req.query.limit) ? 100 : +req.query.limit,
```

Comment:
Maybe this is a convention, but it feels a little sketch to coerce what is likely a bad request into these defaults. Instead of communicating to the user that the query parameter passed was invalid

Reply:
I mean either works,

I would prefer to keep this default here, and instead reject the request in the schema validation if you prefer that.

but at the end of the day the user will know what amount of items we're giving them based on the numResults value in the response, so for me its mostly about making the API easier to use.

Reply:
Yeah to be clear, I'm fine with setting defaults. But yeah please add atleast some basic schema validation. Allowing any string to enter our system sounds like a security vulnerability.

Reply:
Any string cant enter our system, this code is only allowing a number. If the string cannot be parsed into a number it is replaced by the default. This is similarly effective as to putting a check earlier in the validation, because the bad value is being stripped before reaching our service code.

Same with the object properties, if the projection values cannot be matched to a DomainField name it will not be included

Reply:
I agree it's similarly effective, but was concerned that the lack of explicit validation might open up additional attack vectors at the API lvl.  But this is fine for now.